src/OrbitBase/ThreadPool.cpp

```diff
@@ -197,4 +213,34 @@ std::shared_ptr<ThreadPool> ThreadPool::Create(
                                           std::move(run_action));
 }
 
+void ThreadPool::InitializeDefaultThreadPool() { (void)GetDefaultThreadPool(); }
+
+ThreadPool* ThreadPool::GetDefaultThreadPool() {
+  absl::MutexLock lock(&g_default_thread_pool_mutex);
+  if (g_user_default_thread_pool != nullptr) {
+    return g_user_default_thread_pool.get();
+  }
+
+  if (g_default_thread_pool != nullptr) {
+    return g_default_thread_pool.get();
+  }
+
+  const unsigned number_of_logical_cores = std::thread::hardware_concurrency();
+  // std::thread::hardware_concurrency may return 0 on unsupported platforms but that shouldn't
+  // occur on standard Linux or Windows.
+  ORBIT_CHECK(number_of_logical_cores > 0);
+
+  g_default_thread_pool = orbit_base::ThreadPool::Create(
+      /*thread_pool_min_size=*/number_of_logical_cores,
+      /*thread_pool_max_size=*/number_of_logical_cores, /*thread_ttl=*/absl::Seconds(1),
+      /*run_action=*/[](const std::unique_ptr<Action>& action) { action->Execute(); });
+
+  return g_default_thread_pool.get();
+}
+
+void ThreadPool::SetDefaultThreadPool(std::shared_ptr<ThreadPool> thread_pool) {
```

Comment:
Not sure about this: If called multiple times this invalidates pointers previouly handed out by GetDefaultThreadPool. The esiest way around this would be to just go with one default size threadpool (as suggested above).

Reply:
Implemented that approach, see comment above.